### PR TITLE
Don't use too many threads in mapreduce kernel.

### PR DESCRIPTION
**Pull request by maleadt**
Fix https://github.com/JuliaGPU/CuArrays.jl/issues/160
@avik-pal please test

### CODE REVIEW

File: src/mapreduce.jl [hunk 1]
```diff
@@ -54,25 +54,32 @@ function mapreducedim_kernel_parallel(f, op, R::CuDeviceArray{T}, A::CuDeviceArr
...
-    else
-        CIS = CartesianIndices(Ssize)
-        blk, thr = (Rlength - 1) ÷ outer_thr + 1, (inner_thr, outer_thr, 1)
-        @cuda(blocks=blk, threads=thr,
-              mapreducedim_kernel_parallel(f, op, R, A, CIS, Rlength, Slength))
+    CIS = CartesianIndices(Ssize)
+
+    parallel_args = (f, op, R, A, CIS, Rlength, Slength)
+    GC.@preserve parallel_args begin
+        parallel_kargs = Tuple(adapt(CUDAnative.Adaptor(), arg) for arg in parallel_args)
```
[the remark this answers is not in the document]
Comment (maleadt) [reply src/mapreduce.jl]: could just have an alias in CUDAnative, sure.

**Comment by avik-pal** [comment]
The error still persists. I am adding the debug mode output in this [gist](https://gist.github.com/avik-pal/da43ff2ba2d1ac3d9f4644aeb68a6105)

**Comment by maleadt** [comment]
OK, confirmed. The code didn't properly respect the thread limit. Please try again.

**Comment by avik-pal** [comment]
Works great.

**Pull request merged**